graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -249,4 +250,63 @@ static boolean hasExpectedSchema(Table table) {
     Column contents = table.column(CONTENTS_COLUMN_NAME);
     return contents != null && contents.type() == Column.Type.Varchar;
   }
+
+  public Map<String, Object> tryDeployNewSchema(
+      String namespace, Callable<Map<String, Object>> newSchemaCreator) throws Exception {
+    ensureTableExists(namespace);
+    if (transitionToSchemaDeploymentInProgress(namespace)) {
+      try {
+        return newSchemaCreator.call();
+      } finally {
+        transitionToNoSchemaDeploymentInProgress(namespace);
+      }
+    } else {
+      throw new IllegalStateException(
+          "There is a pending schema deployment process. Please retry later.");
+    }
+  }
+  /**
+   * Returns true if there was no schema deployment in progress. It means that that is safe to
+   * deploy a new schema. If it returns false, it means that there is already schema deployment in
+   * progress.
+   */
+  private boolean transitionToSchemaDeploymentInProgress(String namespace)
+      throws ExecutionException, InterruptedException {
+
+    BuiltCondition schemaDeploymentInProgressNull =
+        BuiltCondition.of(DEPLOYMENT_IN_PROGRESS, Predicate.EQ, null);
```

Comment:
And to answer your question further down: you can use
```java
DEPLOYMENT_IN_PROGRESS, Predicate.NEQ, true
```
to deal with the fact that it will be null the first time.